app/controllers/api/v1/lobbies_controller.rb

```diff
@@ -4,18 +4,41 @@ module Api
   module V1
     class LobbiesController < AuthenticatedController
       before_action :require_authentication, except: [:from_code, :join]
-      before_action :require_authorisation, only: [:show, :update, :destroy]
+      before_action :require_authorisation, only: [:show, :update, :destroy, :start]
       before_action :set_quiz, only: [:index, :create]
+      before_action :set_lobby, only: [:join, :start, :answer, :show, :update, :destroy]
+      before_action :set_player, only: [:answer]
+      before_action :set_current_question, only: [:answer]
 
       def join
-        lobby = Lobby.find(params[:id])
-
-        @player = lobby.players.create(player_params)
+        @player = @lobby.players.create(player_params)
 
-        Pusher.trigger(lobby.code, Lobby::PLAYER_JOIN, { id: @player.id })
+        Pusher.trigger(@lobby.code, Lobby::PLAYER_JOIN, { id: @player.id })
         render :player, status: :created
       end
 
+      def start
+        @lobby.status = :in_progress
+
+        if @lobby.save
+          Pusher.trigger(@lobby.code, Lobby::LOBBY_START, {})
+          render :show
+        else
+          render 'api/v1/model_errors', locals: { errors: @lobby.errors }, status: :unprocessable_entity
+        end
+      end
+
+      def answer
+        return head :unauthorized unless @lobby.players.include? @player
+
+        replace_player_answers
+
+        Pusher.trigger(@lobby.code, Lobby::ANSWER_SENT,
+                       { answer_count: @question.player_answers.distinct.count('player_id') })
```

Comment:
For a given `[PlayerAnswer(player_id: 1, answer_id: 1), PlayerAnswer(player_id: 1, answer_id: 2)]`, will `.distinct.count('player_id')` return `1` or `2`?

Reply:
It will return 1. Tested even with three answers.

Reply:
Nice! 👍